bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ImageImpl.java

```diff
@@ -198,6 +202,7 @@ protected void initModel() {
                 baseResourcePath = resource.getPath();
             }
             baseResourcePath = resource.getResourceResolver().map(request, baseResourcePath);
+            final String assetName = getAssetName();
```

Comment:
you need to check getImageName() as that is supposed to overwrite the asset name

Reply:
Sorry, I didn't understand your comment. This method is internally calling `getSeoFriendlyFileName()` which does the overwrite work. I will adjust the comments. Please correct me if there is something else you meant.